src/Core/Api/APIService.php

```diff
@@ -109,11 +130,34 @@ public function query($query, $method = 'post', $variables = [])
           'Body' => $body_for_logging
         ], 'method' => __METHOD__]);
 
-      return $this->client->call($method, $arguments);
+      $response =  $this->client->call($method, $arguments);
```

Comment:
Is the auth failure due to token expired? If so, can we just check if the token expired and refresh the token before firing the call?

Reply:
The `AuthService` module already handles the normal expiration situation.
This draft PR is over a month old though, so I don't suggest investing too much time in reading it. I think I was just experimenting.